mmv1/products/netapp/volume.yaml

```diff
@@ -0,0 +1,389 @@
+    immutable: true
+  - !ruby/object:Api::Type::Boolean
+    name: ldapEnabled
+    description: |
+      Flag indicating if the volume is NFS LDAP enabled or not. Inherited from storage pool.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'activeDirectory'
+    description: |
+      Reports the resource name of the Active Directory policy being used. Inherited from storage pool.
+    output: true
+  - !ruby/object:Api::Type::String
+    name: 'kmsConfig'
+    description: |
+      Reports the CMEK policy resurce name being used for volume encryption. Inherited from storage pool.
+    output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'encryptionType'
+    description: |
+      Reports the data-at-rest encryption type of the volume. Inherited from storage pool.
+    values:
+      - :ENCRYPTION_TYPE_UNSPECIFIED
+      - :SERVICE_MANAGED
+      - :CLOUD_KMS
+    output: true
+  - !ruby/object:Api::Type::Boolean
+    name: 'hasReplication'
+    description: |
+      Indicates whether the volume is part of a volume replication relationship.
+    output: true
+  - !ruby/object:Api::Type::Enum
+    name: 'restrictedActions'
+    description: |
+      List of actions that are restricted on this volume.
+    values:
+      - :RESTRICTED_ACTION_UNSPECIFIED
+      - :DELETE
+  - !ruby/object:Api::Type::Array
+    name: 'mountOptions'
+    description: Reports mount instructions for this volume.
+    output: true
+    item_type: !ruby/object:Api::Type::NestedObject
+      properties:
+        - !ruby/object:Api::Type::String
+          name: 'export'
+          description: Export path of the volume.
+        - !ruby/object:Api::Type::String
+          name: 'exportFull'
+          description: |
+            Full export ptah of the volume.
+
+            Format for NFS volumes: `<export_ip>:/<shareName>`
+
+            Format for SMB volumes: `\\\\netbios_prefix-four_random_hex_letters.domain_name\\shareName`
+        - !ruby/object:Api::Type::String
+          name: 'instructions'
+          description: Human-readable mount instructions.
+        - !ruby/object:Api::Type::Enum
+          name: 'protocol'
+          description: |
+              Protocol to mount with.
+          values:
+            - :PROTOCOLS_UNSPECIFIED
+            - :NFSV3
+            - :NFSV4
+            - :SMB
+  - !ruby/object:Api::Type::NestedObject
+    name: snapshotPolicy
+    description: Snapshot policy defines the schedule for automatic snapshot creation.
+    properties:
+      - !ruby/object:Api::Type::Boolean
+        name: 'enabled'
+        description: |-
+          Enables automated snapshot creation according to defined schedule. Default is false.
+      - !ruby/object:Api::Type::NestedObject
+        name: 'hourlySchedule'
+        description: |-
+          Hourly schedule policy.
+        properties:
+          - !ruby/object:Api::Type::Integer
+            name: 'snapshotsToKeep'
```

Comment:
I would consider making `snapshotsToKeep` required. Terraform can act a little weird when a block is empty, so that'll ensure that it never is, and not specifying a value seems fairly niche.